Fix create PR in release workflow

shashank-elastic opened the pull request:
## Issues
#3527

## Summary

- Use the supported format of GitHub PR for 3.12 
- Per the 3.12 [docs](https://pygithub.readthedocs.io/en/stable/github_objects/Repository.html#github.Repository.Repository.create_pull) the expected format would be
```
pr = repo.create_pull(
    base=base_branch,
    head=branch_name,
    title=message,
    body=body,
    maintainer_can_modify=True,
    draft=draft
)
```

shashank-elastic commented:
The workflow was issued to test from private branch on old backport of 8.9 https://github.com/elastic/detection-rules/actions/runs/8380327773/job/22949293741

But the cahnges are not reflected. Checking this further.

shashank-elastic commented:
The workflow will always point this to the lock-versions commit head so practically this change was not reflected in the workflow run. Should find a way to test this

![image](https://github.com/elastic/detection-rules/assets/91139415/bf190459-0e01-479b-856e-38c3a86be9d8)

Mikaayenson commented:
> The workflow will always point this to the lock-versions commit head so practically this change was not reflected in the workflow run. Should find a way to test this
> 


@shashank-elastic You don't need to run the entire workflow to test. You can debug the method locally that's failing (and pass the correct parameters). 

```bash
python -m detection_rules dev integrations-pr \
    $LOCAL_REPO                                 \
    --github-repo $TARGET_REPO                  \
    --base-branch $TARGET_BRANCH                \
    --assign shashank-elastic                  \
    $DRAFT_ARGS
```

I recommend setting up a debug session and set breakpoints in the `integrations-pr` command.

shashank-elastic commented:
**Local Testing**

- Build a test release package
- Installed go "elastic-package" as a dependancy 
- Issued a local command to create the PR 

```
python -m detection_rules dev integrations-pr ../integrations --base-branch test-dr-release-3527 --assign shashank-elastic

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

Switched to a new branch 'test-dr-release-3527'
From github.com:elastic/integrations
 * branch                test-dr-release-3527 -> FETCH_HEAD
Switched to a new branch 'detection-rules/8.14.1-beta.1-52868768'
2024/03/26 22:09:53  WARN CommitHash is undefined, in both /Users/shashankks/.elastic-package/version and the compiled binary, config may be out of date.
2024/03/26 22:09:53  WARN reading version file failed: open /Users/shashankks/.elastic-package/latestVersion: no such file or directory
Format the package
Done
Enumerating objects: 11, done.
Counting objects: 100% (11/11), done.
Delta compression using up to 12 threads
Compressing objects: 100% (6/6), done.
Writing objects: 100% (6/6), 558 bytes | 558.00 KiB/s, done.
Total 6 (delta 5), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (5/5), completed with 5 local objects.
remote: 
remote: Create a pull request for 'detection-rules/8.14.1-beta.1-52868768' on GitHub by visiting:
remote:      https://github.com/elastic/integrations/pull/new/detection-rules/8.14.1-beta.1-52868768
remote: 
remote: GitHub found 1 vulnerability on elastic/integrations's default branch (1 moderate). To find out more, visit:
remote:      https://github.com/elastic/integrations/security/dependabot/16
remote: 
To github.com:elastic/integrations.git
 * [new branch]          detection-rules/8.14.1-beta.1-52868768 -> detection-rules/8.14.1-beta.1-52868768
PR created:
https://github.com/elastic/integrations/pull/9452
2024/03/26 22:10:00  WARN CommitHash is undefined, in both /Users/shashankks/.elastic-package/version and the compiled binary, config may be out of date.
Format the package
Done
2024/03/26 22:10:01  WARN CommitHash is undefined, in both /Users/shashankks/.elastic-package/version and the compiled binary, config may be out of date.
Lint the package
Done
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Delta compression using up to 12 threads
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 459 bytes | 459.00 KiB/s, done.
Total 5 (delta 4), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
remote: 
remote: GitHub found 1 vulnerability on elastic/integrations's default branch (1 moderate). To find out more, visit:
remote:      https://github.com/elastic/integrations/security/dependabot/16
remote: 
To github.com:elastic/integrations.git
   49d332673..ee2d78ea0  detection-rules/8.14.1-beta.1-52868768 -> detection-rules/8.14.1-beta.1-52868768
(.venv) 
detection-rules on  issue-3527 [$!?] is 📦 v0.1.0 via 🐹 v1.22.1 via 🐍 v3.12.2 (.venv) on ☁️  shashank.suryanarayana@elastic.co took 21s 
❯ 
```

PR successfully created - https://github.com/elastic/integrations/pull/9452

Mikaayenson commented on detection_rules/etc/packages.yml:
```diff
@@ -18,13 +18,13 @@ package:
       github: elastic/protections
       type: elastic
+    source:
+      license: Elastic-2.0
     title: Prebuilt Security Detection Rules
     type: integration
-    version: 8.14.0-beta.0
+    version: 8.14.1-beta.1
```
Why are these changes in here?

shashank-elastic replied:
Reverted them was generated when things were tested locally

eric-forte-elastic commented:
Just to double check, should the [other call](https://github.com/elastic/detection-rules/pull/3528/files#diff-e451accac6e3d163c27239dce087f27f707daf31a84a45b12ebd0f0c26dc8c83R523-R524) be updated to match this one as well?

```python
    pr = repo.create_pull(title, body, base=kwargs["base_branch"], head=branch_name, maintainer_can_modify=True,
                          draft=draft)
```

shashank-elastic commented:
> Just to double check, should the [other call](https://github.com/elastic/detection-rules/pull/3528/files#diff-e451accac6e3d163c27239dce087f27f707daf31a84a45b12ebd0f0c26dc8c83R523-R524) be updated to match this one as well?
> 
> ```python
>     pr = repo.create_pull(title, body, base=kwargs["base_branch"], head=branch_name, maintainer_can_modify=True,
>                           draft=draft)
> ```

Honestly, that was not scoped here, so if we intend to that we would need to test the workflow kibana-pr. 
This would mean understanding what it takes to create a kibana-pr and test. But this fix needs to go before next week DR release else we have the release automation workflow failures like we saw in 8.12.2 release. 
If we can decouple the fixes and send this in given the time frame we are operating it would be beneficial! 

Also looking at this [post](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request) title and body are not the "required parameters" so may be we will no hit the issue like in integrations_pr where even the required parameters such as "base_branch" and "branch_name" are also not aligned as per 3.12 requirements.  

cc @eric-forte-elastic

eric-forte-elastic reviewed (approved):
Thanks for the [explanation](https://github.com/elastic/detection-rules/pull/3528#issuecomment-2021123681)! Also looks like the kibana pr dev command is no longer used in release workflows so not required to fix for the release process. 

LGTM :+1:

Mikaayenson commented:
[Concur](https://github.com/elastic/detection-rules/pull/3528#discussion_r1534972348), with @eric-forte-elastic. The other references should be updated.

shashank-elastic commented:
@Mikaayenson The deprectaion is handled in https://github.com/elastic/detection-rules/pull/3552.
Can we push this async.

Mikaayenson reviewed (approved):
LGTM
